include/envoy/filesystem/filesystem.h

```diff
@@ -61,6 +63,11 @@ class File {
    * @return string the file path
    */
   virtual std::string path() const PURE;
+
+  /**
+   * @return the type of the destination
+   */
+  virtual DestinationType destinationType() const PURE;
```

Comment:
Is this accessor needed?

Reply:
It is used for testing to ensure correctness